redshift_sqlalchemy/dialect.py

```
@@ -320,67 +323,157 @@ def visit_unload_from_select(element, compiler, **kw):
+        the `manifest` option is used
+    credentials : str
+        AWS credentials of the form
+        aws_access_key_id=<access-key-id>;aws_secret_access_key=<secret-access-key>[;token=<temporary-session-token>]
+    delimiter : File delimiter, optional
+        defaults to ','
+    ignore_header : int, optional
+        Integer value of number of lines to skip at the start of each file
+    null : str, optional
+        Optional string value denoting what to interpret as a NULL value from
+        the file
+    manifest : bool, optional
+        Boolean value denoting whether data_location is a manifest file.
+    empty_as_null : bool, optional
+        Boolean value denoting whether to load VARCHAR fields with empty
+        values as NULL instead of empty string
+    blanks_as_null : bool, optional
+        Boolean value denoting whether to load VARCHAR fields with whitespace
+        only values as NULL instead of whitespace
+    format : str, optional
+        CSV, JSON, or AVRO. Indicates the type of file to copy from.
+    compression : str, optional
+        GZIP, LZOP, indicates the type of compression of the file to copy
+    """
+    def __init__(self, table, data_location, credentials,
+                 delimiter=',', ignore_header=0, null=r'\N', manifest=False,
+                 empty_as_null=True, blanks_as_null=True,
+                 format='CSV', compression=None):
+        # regex from here: http://blogs.aws.amazon.com/security/blog/tag/key+rotation
+        if not creds_rx.match(credentials):
+            raise ValueError('"credentials" parameter must match the following'
+                             ' format:\n'
+                             'aws_access_key_id=<access-key-id>;'
+                             'aws_secret_access_key=<secret-access-key>'
+                             '[;token=<temporary-session-token>]\ngot %r' %
+                             credentials)
```

this isn't brittle, this simply fails fast with invalid data.